aiomysql/pool.py

```diff
@@ -125,6 +125,10 @@ def acquire(self):
                     conn = self._free.popleft()
                     assert not conn.closed, conn
                     assert conn not in self._used, (conn, self._used)
+                    # drop if connection timedout
```

Comment:
Would you move the patch into `_fill_free_pool()` method?
It makes code more clear from my perspective.
Just iterate over `_free` list and drop all timed-out connections.